heat/core/devices.py

```diff
@@ -74,6 +74,31 @@ def __str__(self) -> str:
         """
         return "{}:{}".format(self.device_type, self.device_id)
 
+    def __eq__(self, other: Any) -> bool:
+        """
+        Overloads the `==` operator for local equal check.
+
+        Parameters
+        ----------
+        other : Any
+            The object to compare with
+        """
+        if isinstance(other, Device):
+            return (
+                True
+                if (self.device_type == other.device_type and self.device_id == other.device_id)
```

Comment:
You can shorten this to a return of the condition, i.e. return self.device_type == other...